src/number_theory/pythagorean_triples.lean

```diff
@@ -505,25 +514,10 @@ begin
       { rw [pow_two, ← h.left.eq], ring },
       simpa using eq_or_eq_neg_of_pow_two_eq_pow_two _ _ this } },
   { rintro ⟨m, n, ⟨rfl, rfl⟩ | ⟨rfl, rfl⟩, rfl | rfl, co, pp⟩;
-    delta pythagorean_triple,
-    { split, { ring },
-      cases pp with pp pp,
-      { apply coprime_pow_two_sub_mul_of_even_odd co pp.left pp.right },
-      { apply coprime_pow_two_sub_mul_of_odd_even co pp.left pp.right } },
-    { split, { ring },
-      cases pp with pp pp,
-      { apply coprime_pow_two_sub_mul_of_even_odd co pp.left pp.right },
-      { apply coprime_pow_two_sub_mul_of_odd_even co pp.left pp.right } },
-    { split, { ring },
-      rw int.gcd_comm,
-      cases pp with pp pp,
-      { apply coprime_pow_two_sub_mul_of_even_odd co pp.left pp.right },
-      { apply coprime_pow_two_sub_mul_of_odd_even co pp.left pp.right } },
-    { split, { ring },
-      rw int.gcd_comm,
-      cases pp with pp pp,
-      { apply coprime_pow_two_sub_mul_of_even_odd co pp.left pp.right },
-      { apply coprime_pow_two_sub_mul_of_odd_even co pp.left pp.right } } }
+    delta pythagorean_triple;
```

Comment:
If I understand the proof correctly, you have two goals before this `delta`. So I would just write a `,` instead of a `;`.
Because in the end you are proving the two goals separately anyway. Then you can also replace the `<|>` with a simple `,`.
You can move the `delta` before the `rintro` if you want to deduplicate it.

Reply:
The `rintro` creates 4 goals (the two cases for `x` and `y` and the two cases of parity of `m` and `n`). I could do `delta, rintro;` instead of `rintro; delta;`. Maybe that is simpler?

Reply:
Ooh, sorry... I didn't look carefully. You are absolutely right.
Yes, please move the delta before the rintro, and then this is fine.
LGTM!